File: pyWorks/ch05/passed_time.py
# 지나온 날짜 계산하기: 종료일(오늘) - 시작일(과거) = 일수
# datetime.date(종료일) - datetime.date(시작일)
# datetime.date(특정일).weekday(): 요일(인덱스) - 월요일이 0번
import datetime
import calendar as cal

print('♠ 지금까지 몇 일? ♠')
day1 = datetime.date(2023, 3, 16)
print(day1)

# today = datetime.date(2023, 4, 21)
today = datetime.date.today()
print(today)

passed_time = today - day1

# print(passed_time)
print('개강 이후 {}일이 지났습니다.'.format(passed_time.days))

# 날짜로 요일 알아내기
days = ['월', '화', '수', '목', '금', '토', '일']

# weekday() - 요일 나타내는 함수
the_day = datetime.datetime(2023, 3, 16).weekday()

# print(the_day)
print(f'{days[the_day]}요일')

now = today.weekday()

# print(now)
print(f'{days[now]}요일')

# 2023. 3월 출력
cal.prcal(2023)
cal.prmonth(2023, 3)

File: javaWorks/ch03/src/datetime/PassedTime.java
package datetime;

import java.util.Calendar;

public class PassedTime {

	public static void main(String[] args) {
		
		Calendar today = Calendar.getInstance();
		Calendar theDay = Calendar.getInstance();
		
		// System.out.println(today);
		// System.out.println(today.get(Calendar.YEAR));
		// System.out.println(today.get(Calendar.MONTH) + 1);
		// System.out.println(today.getTimeInMillis());
		
		// 년으로 환산
		// System.out.println(today.getTimeInMillis() / (24*60*60*1000)/365);
		// 일로 환산
		// System.out.println(today.getTimeInMillis() / (24*60*60*1000));
		
		// 시작일과 종료일 지정
		theDay.set(2023, 3, 16);	// 시작일
		today.set(2023, 5, 24);		// 시작일
		
		// 날짜 출력 메소드
		System.out.println(datePrint(theDay));
		System.out.println(datePrint(today));
		
		// 날짜 출력
		String startDay = theDay.get(Calendar.YEAR) + "년 "
				+ theDay.get(Calendar.MONTH) + "월 " + theDay.get(Calendar.DATE) + "일";
		String endDay = today.get(Calendar.YEAR) + "년 "
				+ today.get(Calendar.MONTH) + "월 " + today.get(Calendar.DATE) + "일";
		// System.out.println(startDay);
		// System.out.println(endDay);
		
		// 지나온 시간 계산
		long passedTime = (today.getTimeInMillis() - theDay.getTimeInMillis());
		// System.out.println(passedTime);
		
		// 일로 환산
		long passedDay = passedTime / (24*60*60*1000);
		System.out.println("개강 이후 " + passedDay + "일이 지났습니다.");
		
	}
	
	public static String datePrint(Calendar date) {	// 날짜 객체를 매개로 전달받음
		
		String day = date.get(Calendar.YEAR) + "년 "
				+ date.get(Calendar.MONTH) + "월 " + date.get(Calendar.DATE) + "일";
		
		return day;
		
	}

}
